lib/pipelineTemplateVersionFactory.js

```diff
@@ -203,38 +204,48 @@ class PipelineTemplateVersionFactory extends BaseFactory {
     /**
      * Get a version of a pipeline template and template meta based on the criteria
      * @method getWithMetadata
-     * @param  {Object}     config               Config object
-     * @param  {Datastore}  config.datastore     Object that will perform operations on the datastore
-     * @param  {String}     [config.templateId]  The template id
-     * @param  {String}     config.name          The template name
-     * @param  {String}     config.namespace     The template namespace
-     * @param  {String}     config.version       The template version
-     * @param  {Factory}   templateMetaFactory  Template meta factory
-     * @return {Promise}                         {PipelineTemplateVersionModel, PipelineTemplateMetaModel}
+     * @param  {Object}     config                Config object
+     * @param  {String}     config.name           The template name
+     * @param  {String}     config.namespace      The template namespace
```

Comment:
Factory should not be designed for a specific need.
We should allow accepting `templateId`.